### DISCUSSION TITLE
using template repositories via ssh: how to configure `ssh-agent`

### DISCUSSION THREAD

**Discussion by admin** [cargo-generate/cargo-generate#653] [Q&A]
## Windows / PowerShell

1. Enable the `ssh-agent` service
```sh
## Run this as Adminstrator

# By default the ssh-agent service is disabled. Allow it to be manually started for the next step to work.
Get-Service ssh-agent | Set-Service -StartupType Manual
```

2. start the `ssh-agent` service and add your key
```sh
# Start the service
Start-Service ssh-agent

# This should return a status of Running
Get-Service ssh-agent

# Now load your key files into ssh-agent
ssh-add ~\.ssh\id_ed25519
```

[1]: https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_keymanagement

**Comment by sassman** [comment]
on windows there are still issues with using ssh-agent on non PowerShells. See https://github.com/libgit2/libgit2/issues/4958

Please use the OpenSSH ssh-agent started via PowerShell, not the git-bash style with the `SSH_AUTH_SOCK`. It does not reliably work.
